### ISSUE TITLE
Responsive Redesign // KB Article Pages

### ISSUE THREAD

**Issue by jess-cook03** [mozilla/sumo-project#55]
[Article page breakdown](https://docs.google.com/spreadsheets/d/1DmVBJxHIgoRYZUWKzxIqw1rUjUXz0wV2amtm1-a9jKY/edit?usp=sharing)

**Comment by jess-cook03** [comment]
@kkellydesign updated [zeplin link](https://app.zeplin.io/project/5db3524d3cd0612c5d2f78a9/dashboard?seid=5ddf082ec7290905ff852cef) -- note: sticky feedback component and search in navigation

**Comment by kkellydesign** [comment]
@jess-cook03 I have this sticky component done in the sidebar, and you'll be able to see it when this goes live: https://github.com/mozilla/kitsune/pull/4150

I want to push back a little on making it sticky on mobile though. At what point do you want it to pop up? I think if we put it anywhere other than inline at the bottom of the content area, it'll be more likely to be seen as an interruption and people will close it. If we put it at the bottom of the content area, that's about when people are going to be done scrolling anyway, so at that point, I'm not sure there's much advantage to making it persist on scroll. What do you think?

**Comment by jess-cook03** [comment]
@kkellydesign per our discussion this morning -- design is cleared to make this change. Sticky component is only on desktop/large viewports and all smaller viewports get in the in-page, end of article module that asks if the article is helpful. Therefore confirming new rule -- no sticky component on smaller breakpoints

**Comment by kkellydesign** [comment]
This is ready for QA: https://test.sumo.mozit.cloud/en-US/kb/how-do-i-tell-if-32-bit-or-64-bit

**Comment by ovidiuboca1** [comment]
cc @jess-cook03 @kkellydesign 
I have some questions:
1. I tested this on desktop Mac OS X 10.14 and WIn 10 and the "Related Articles" section is not rendered correctly on both browsers Firefox and Chrome. Please see the print screen with the AR and let me know your opinion. 
![Screenshot 2020-02-14 at 14 44 11](https://user-images.githubusercontent.com/54845678/74532681-7ea92a80-4f38-11ea-85ec-6ee1604e7135.png)
2. On mobile devices, the picture that is in the User-Agent section is not displayed comparing it with the desktop. I made a print screen on desktop with the picture that is missing on mobile. 
![Screenshot 2020-02-14 at 14 41 09](https://user-images.githubusercontent.com/54845678/74533013-3dfde100-4f39-11ea-94ae-3ace02b043a2.png)
3. I saw in the last comment from jess that the desired behavior for smaller viewports is to have at the end of the article a module that asks if the article is helpful. I did a print screen with the actual result on the mobile device. 
![Skype_Picture_2020_02_14T12_53_48_103Z](https://user-images.githubusercontent.com/54845678/74533255-d300da00-4f39-11ea-808a-961c5b2f11f4.jpeg) My question is, this is the desired design?

**Comment by kkellydesign** [comment]
@ovidiuboca1 That's not the correct mobile design, but it should be correct now with the most recent updates. Please try that one again. 

Otherwise, I just fixed the related articles issue. That will go up with https://github.com/mozilla/kitsune/pull/4177

**Comment by kkellydesign** [comment]
The mobile layout and Related Articles fix is ready for QA Here's an example: https://test.sumo.mozit.cloud/en-US/kb/firefox-osx

**Comment by ovidiuboca1** [comment]
I verified this and I can confirm the fix.

**Comment by jess-cook03** [comment]
@kkellydesign @ovidiuboca1 
Finding a couple bugs/edits here //

- [ ] **BUG** non-logged in users are seeing the article editing capabilities, see image below
![Screen Shot 2020-02-24 at 1 13 57 PM](https://user-images.githubusercontent.com/54991698/75185881-d4d65480-5714-11ea-8781-5614ae966408.png)

- [ ] 'Is this article helpful' component --- when you up-vote nothing happens -- when you down-vote nothing happens.... ??
The site today functions like this --->
![Screen Shot 2020-02-24 at 2 50 23 PM](https://user-images.githubusercontent.com/54991698/75185996-1109b500-5715-11ea-969e-9c549ff1a56d.png)
![Screen Shot 2020-02-24 at 2 50 38 PM](https://user-images.githubusercontent.com/54991698/75186003-149d3c00-5715-11ea-9d69-75041df990ab.png)
Can this functionality still happen within our new widget component ?
Like this --->
![Screen Shot 2020-02-24 at 3 05 34 PM](https://user-images.githubusercontent.com/54991698/75187007-2253c100-5717-11ea-8a63-b6d03fbbf36b.png)

**Comment by kkellydesign** [comment]
> 'Is this article helpful' component --- when you up-vote nothing happens -- when you down-vote nothing happens.... ?? The site today functions like this

I see that. This message lives in wiki/views.py. @akatsoulas Can you point me in the right direction for where this frontend code lives? There are two things going on here – one is the tooltip immediate feedback, and the other is the modal with the request for more info. The modal works correctly on the /question threads, but the form on the wiki pages just disappears without any feedback, and the page refreshes. For example: 

https://test.sumo.mozit.cloud/en-US/kb/scan-qr-codes-firefox-android

This is a backend form submission, so I'm not sure if the javascript is supposed to hijack this and submit the form with ajax.

Anyway, if you could get me halfway there, I can make it look good.

**Comment by akatsoulas** [comment]
@kkellydesign the frontend code for this lives here
https://github.com/mozilla/kitsune/blob/master/kitsune/wiki/jinja2/wiki/includes/document_macros.html#L305

**Comment by kkellydesign** [comment]
@akatsoulas yes, I've already edited that form to look like the new design. The issue is what happens when the form is submitted. Jess is asking for the feedback functionality to be different, and replace that form with either a 'thank you' message or an additional form that requests more info. See the part of the comment above that starts here: 
> Can this functionality still happen within our new widget component ?
Like this --->

**Comment by kkellydesign** [comment]
@akatsoulas @jess-cook03 this is in here: https://github.com/mozilla/kitsune/pull/4201

<img width="414" alt="Screen Shot 2020-03-02 at 10 58 02 AM" src="https://user-images.githubusercontent.com/202590/75698546-ec658e00-5c7c-11ea-9c8e-d4cf73d564b8.png">
<img width="333" alt="Screen Shot 2020-03-02 at 10 59 18 AM" src="https://user-images.githubusercontent.com/202590/75698548-ecfe2480-5c7c-11ea-9489-8f8add412b32.png">
<img width="367" alt="Screen Shot 2020-03-02 at 11 00 52 AM" src="https://user-images.githubusercontent.com/202590/75698550-ecfe2480-5c7c-11ea-9c11-9b4170a0c777.png">
<img width="783" alt="Screen Shot 2020-03-02 at 11 04 27 AM" src="https://user-images.githubusercontent.com/202590/75698551-ed96bb00-5c7c-11ea-9376-fbce2930f68f.png">
<img width="859" alt="Screen Shot 2020-03-02 at 11 53 58 AM" src="https://user-images.githubusercontent.com/202590/75698555-ed96bb00-5c7c-11ea-9c43-b47f512373d1.png">

**Comment by ovidiuboca1** [comment]
I verified this on Windows 10 and Mac OS X 10.14 with Firefox v73 and Chrome and I can confirm the fixes. 
@jess-cook03 do you think we can close this and move it to ready to deploy?

**Comment by jess-cook03** [comment]
Good to go! @ovidiuboca1
